js/analyzer.js

```diff
@@ -1,65 +1,232 @@
+    if(typeof this.config.queue != 'undefined' && this.config.queue.length != 0){
+        this.queue = this.config.queue;
+    }else{
+        this.queue = ['keywordDensity', 'subheaderChecker'];
+    }
+    //if no available keywords, load default array
+    if(typeof this.config.wordsToRemove == 'undefined'){
+        this.config.wordsToRemove =  [' a', ' in', ' an', ' on', ' for', ' the', ' and'];
+    }
+    //set default variables
+    this.keywordRegex = new RegExp(this.config.keyword);
+    this._output = [];
+}
+
+/**
+ * starts queue of functions executing the analyzer functions untill queue is empty.
+ */
+Analyzer.prototype.runQueue = function(){
+    //remove first function from queue and execute it.
+    if(this.queue.length > 0){
+        this._output.push(this[this.queue.shift()]());
+        this.runQueue();
+    }
+};
+/**
+ * clears current queue of functions, effectively stopping execution of the analyzer.
+ */
+Analyzer.prototype.abortQueue = function(){
+    //empty current Queue
+    this.queue = [];
+};
+
+
+/**
+ * checks the keyword density of given keyword against the cleantext stored in _store.
+ * @returns resultObject
+ */
+Analyzer.prototype.keywordDensity = function(){
+    if(yst_pp._store.wordcount >= 100) {
+        var keywordMatches = yst_pp._store.cleanText.match(new RegExp(this.config.keyword, 'g'));
+        if ( keywordMatches != null ) {
+            var keywordCount = keywordMatches.length;
+            var keywordDensity = (keywordCount / yst_pp._store.wordcount - (keywordCount - 1 * keywordCount)) * 100;
+        } else {
+            var keywordDensity = 0;
+        }
+        switch (true) {
+            case (keywordDensity < 1):
+                var rating =  4;
+                break;
+            case (keywordDensity > 4.5):
+                var rating = -50;
+                break;
+            default:
+                var rating = 9;
+                break;
+        }
+        result = {name: 'keywordDensity', result: {keywordDensity: keywordDensity.toFixed(1) }, rating: rating};
+    }else{
+        result = {name: 'keywordDensity', result: null, rating: null};
+    }
+    return result;
+};
+/**
+ * checks if keywords appear in subheaders of stored cleanTextSomeTags text.
+ * @returns resultObject
+ */
+Analyzer.prototype.subheaderChecker = function() {
+
+    //regex for headers
+    var headers = yst_pp._store.cleanTextSomeTags.match(/<h([1-6])(?:[^>]+)?>(.*?)<\/h\1>/g);
+    if(headers == null){
+        result = {name: 'subHeadings', result: null, rating: 7};
+    }else {
+        var foundInHeader = 0;
+        for (var i = 0; i < headers.length; i++) {
+            var formattedHeaders = this.stripKeywords(headers[i]);
+            if (formattedHeaders.match(new RegExp(this.config.keyword, 'g')) || headers[i].match(new RegExp(this.config.keyword, 'g'))) {
+                foundInHeader++;
+            }
+        }
+        if (foundInHeader > 0) {
```

Comment:
Please move the rating to a different function, so the functions have a single responsibility. In the case of subhheaderChecker I would expect a function that checks the subheader. The subheader rating would be a different function.